node/forest_libp2p/src/service.rs

```diff
@@ -368,32 +348,41 @@ where
     }
 }
 async fn emit_event(sender: &Sender<NetworkEvent>, event: NetworkEvent) {
-    if !sender.is_full() {
-        sender.send(event).await
-    } else {
-        // TODO this would be better to keep the events that would be ignored in some sort of buffer
-        // so that they can be pulled off and sent through the channel when there is available space
-        error!("network sender channel was full, ignoring event");
+    if sender.send(event).await.is_err() {
+        error!("Failed to emit event: Network channel receiver has been dropped");
     }
 }
 
 /// Builds the transport stack that LibP2P will communicate over
 pub fn build_transport(local_key: Keypair) -> Boxed<(PeerId, StreamMuxerBox), Error> {
     let transport = libp2p::tcp::TcpConfig::new().nodelay(true);
+    let transport = libp2p::websocket::WsConfig::new(transport.clone()).or_transport(transport);
     let transport = libp2p::dns::DnsConfig::new(transport).unwrap();
-    let dh_keys = noise::Keypair::<noise::X25519Spec>::new()
-        .into_authentic(&local_key)
-        .expect("Noise key generation failed");
-    let mut yamux_config = yamux::Config::default();
-    yamux_config.set_max_buffer_size(16 * 1024 * 1024);
-    yamux_config.set_receive_window(16 * 1024 * 1024);
+
+    let auth_config = {
+        let dh_keys = noise::Keypair::<noise::X25519Spec>::new()
+            .into_authentic(&local_key)
+            .expect("Noise key generation failed");
+
+        noise::NoiseConfig::xx(dh_keys).into_authenticated()
+    };
+
+    let mplex_config = {
+        let mut mplex_config = mplex::MplexConfig::new();
+        mplex_config.max_buffer_len(usize::MAX);
+        // mplex_config.set_max_buffer_size(usize::MAX);
```

Comment:
Did you forget to remove this comment?

Reply:
ah ya, it was what the type changed in future libp2p version, I commented for easier switching, but #928 exists